cfe_internal/core/watchdog/templates/watchdog.mustache

```diff
@@ -1,8 +1,320 @@
+      echo "- cf-check observed critical integrity issues" >> ${COLLECTION_REPORT}
+      echo "  #+begin_example\n${observation}\n#+end_example" >> ${COLLECTION_REPORT}
+    fi
+fi
+
+if [ "${PATHOLOGY_COUNT}" -gt "${PATHOLOGY_THRESHOLD}" ]; then
+
+    echo "$(date) Found ${PATHOLOGY_COUNT} symptoms, threshold (${PATHOLOGY_THRESHOLD}) breached." | tee -a ${LOGFILE}
+    echo "* Observations" >> ${COLLECTION_REPORT}
+
+    ps auxwww > "${COLLECTION_DIR}/ps_auxwww.txt"
+    echo "- [[./ps_auxwww.txt][~ps auxwww~]]" >> ${COLLECTION_REPORT}
+
+    ps -elf > "${COLLECTION_DIR}/ps_-elf.txt"
+    echo "- [[./ps_-elf.txt][~ps -elf~]]" >> ${COLLECTION_REPORT}
+
+    find "${CFENGINE_WORKDIR}" >> "${COLLECTION_DIR}/find__var_cfengine.txt"
+    echo "- [[./find__var_cfengine.txt][=/var/cfengine= file list]] - Before remediation" >> ${COLLECTION_REPORT}
+
+    tar -c -f "${COLLECTION_DIR}/sys.workdir-before-remediation.tar" -C "${CFENGINE_WORKDIR}" state outputs
+    echo "- [[./sys.workdir-before-remediation.tar][=sys.workdir-before-remediation.tar=]] - CFEngine WORKDIR artifacts before remediation" >> ${COLLECTION_REPORT}
+
+    ps -efl | grep cf- > "${COLLECTION_DIR}/cf-procs.txt"
+    echo "- [[./cf-procs.txt][~ps -efl | grep cf-~]]" >> ${COLLECTION_REPORT}
+
+    echo "$(date) Initiating apoptosis" | tee -a ${LOGFILE}
+    while IFS= read -r proc
+    do
+        _PID=$(echo $proc | awk '{print $4}')
+        _PROC=$(echo $proc | awk '{print $15}')
+        _PROCFILE=$(echo "$_PROC" | sed 's./._.g')
+        _COREFILE=$(printf "%s_%s.core" "$_PID" "$_PROCFILE")
+        _COREPATH=$(printf "%s/%s" "${COLLECTION_DIR}" "$_COREFILE")
+        _DBX_ERR_LOG=$(printf "%s/%s.dbx.err" "${COLLECTION_DIR}" "$_COREFILE")
+
+        gencore "$_PID" "$_COREPATH" > "${COLLECTION_DIR}/gencore-$_PID.output" 2>&1
+        echo "- [[./gencore-$_PID.output][output from gencore $_PID $_COREPATH]]" >> ${COLLECTION_REPORT}
+        echo "- [[./$_COREFILE][core from $_PID]]" >> ${COLLECTION_REPORT}
+
+        observation=$(echo "where" | dbx "$_PROC" "$_COREPATH" 2> "$_DBX_ERR_LOG")
+        echo "  - backtrace:" >> ${COLLECTION_REPORT}
+        echo "    #+begin_example\n${observation}\n#+end_example" >> ${COLLECTION_REPORT}
+        echo "  - dbx stderr: [[./$(basename $_DBX_ERR_LOG)][dbx $_PROC $_COREPATH]]" >> ${COLLECTION_REPORT}
+        kill -9 "$_PID" > "${COLLECTION_DIR}/kill_$_PID.txt" 2>&1
+        echo "  - [[./kill_$_PID.txt][~kill -9 $_PID~]]" >> ${COLLECTION_REPORT}
+
+    done < "${COLLECTION_DIR}/cf-procs.txt"
+
+    echo "- Purged outputs (don't want them to trigger pathology remediation more than once)" >> ${COLLECTION_REPORT}
+    for each in $(ls -A "${CFENGINE_WORKDIR}/outputs/"); do
+        rm "${CFENGINE_WORKDIR}/outputs/${each}"
+    done
+
+    # Switch to more intelligent cf-check repair in 3.12.3
+    echo "- Purged LMDBs" >> ${COLLECTION_REPORT}
+    for each in $(ls ${CFENGINE_WORKDIR}/state/*.lmdb*); do
```

Comment:
No need to use `ls` here, just use the glob.

Reply:
ack